src/make-gecko-action-hooks.js

```diff
@@ -14,61 +17,94 @@ module.exports.setup = (program) => {
+            '[taskcluster-admin](https://github.com/taskcluster/taskcluster-admin).',
+          ].join('\n'),
+          owner: 'taskcluster-notifications@mozilla.com',
+          emailOnError: false, // true, TODO
+        },
+        schedule: [],
+        task,
+        triggerSchema,
+      });
 
-  for (let action of ACTION_HOOKS) {
-    const hookGroupId = `project-${action.trustDomain}`;
-    const hookId = `in-tree-action-${action.level}-${action.actionPerm}`;
-    const projName = action.trustDomain === 'gecko' ? 'mozilla-central' : 'comm-central';
-    const {task, triggerSchema} = makeHookDetails(taskclusterYml[projName], action);
-    await editHook({
-      noop: options.noop,
-      hookGroupId,
-      hookId,
-      metadata: {
-        name: `Action task ${action.level}-${action.actionPerm}`,
+      // make the role with scopes assume:repo:<repo>:action:<actionPerm> for each repo at this level
+      const projectsAtLevel = Object.keys(projects)
+        .filter(p => projects[p].access === `scm_level_${action.level}`)
+        .map(p => projects[p]);
+      const scopes = projectsAtLevel.map(
+        project => `assume:repo:hg.mozilla.org/${hgmoPath(project)}:action:${action.actionPerm}`);
+      await editRole({
+        roleId: `hook-id:${hookGroupId}/${hookId}`,
         description: [
           '*DO NOT EDIT*',
           '',
-          'This hook is configured automatically by',
+          'This role is configured automatically by',
           '[taskcluster-admin](https://github.com/taskcluster/taskcluster-admin).',
         ].join('\n'),
-        owner: 'taskcluster-notifications@mozilla.com',
-        emailOnError: false, // true, TODO
-      },
-      schedule: [],
-      task,
-      triggerSchema,
-    });
-
-    // make the role with scopes assume:repo:<repo>:action:<actionPerm> for each repo at this level
-    const projectsAtLevel = Object.keys(projects)
-      .filter(p => projects[p].access === `scm_level_${action.level}`)
-      .map(p => projects[p]);
-    const scopes = projectsAtLevel.map(
-      project => `assume:repo:hg.mozilla.org/${hgmoPath(project)}:action:${action.actionPerm}`);
-    await editRole({
-      roleId: `hook-id:${hookGroupId}/${hookId}`,
-      description: [
-        '*DO NOT EDIT*',
-        '',
-        'This role is configured automatically by',
-        '[taskcluster-admin](https://github.com/taskcluster/taskcluster-admin).',
-      ].join('\n'),
-      scopes,
-      noop: options.noop,
-    });
+        scopes,
+        noop: options.noop,
+      });
+    }
   }
 };
 
+const hashTaskclusterYmls = async (projects) => {
+  console.log(chalk.yellow.bold('fetching and hashing `.taskcluster.yml` files from all repos'));
+
+  const result = {};
+  await Promise.all(Object.entries(projects).map(async ([alias, {repo, features}]) => {
+    // try repos don't have a `default` so there's nothing to do
+    if (alias.startsWith('try') || alias.endsWith('try')) {
```

Comment:
Maybe the should be part of the config, rather than looking at the name?

Reply:
I agree - I'd like to solve that another time :)